common/compat.c

```diff
@@ -1037,6 +1038,33 @@ fdwalk (int (* cb) (void *data, int fd),
 
 #endif /* OS_UNIX */
 
+void
+p11_strerror_r (int errnum,
+		char *buf,
+		size_t buflen)
+{
+	intptr_t r = 0;
+
+	assert (buf != NULL);
+	assert (buflen != 0);
+
+	buf[0] = 0xff;
+	errno = 0;
+	r = (intptr_t)strerror_r (errnum, buf, buflen);
+
+	if (r == 0) // XSI success, GNU should never return NULL
```

Comment:
IMO, it would be better to check the behavior difference at compile time at run time. For Meson you could use [compiler.run](https://mesonbuild.com/Compiler-properties.html#compile-and-run-test-application), for Autotools maybe [AC_RUN_IFELSE](https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.71/html_node/Runtime.html) could be used.